### ISSUE TITLE
WixBuild (v4) task fails

### ISSUE THREAD

**Issue by dkalbertson** [wixtoolset/issues#7035]
* Which version of WiX are you building with?

> 4.0.0-preview.1 (I also tried the latest build 4.0.0-rc.1-build.12)

* Which version of Visual Studio are you building with (if any)?

> Visual Studio 2022 17.4.1

* Which version of the WiX Toolset Visual Studio Extension are you building with (if any)?

> Heatwave 0.9.0.5

* Which version of .NET are you building with?

> 4.7.2

* Describe the problem and the steps to reproduce it.

> I used HeatWave to convert a WixLib project from v3 to v4. After conversion, I attempted to build the project. I received the following error out of the WixBuild task:
Error	MSB6004	The specified task executable location "C:\Users\<MaskedUserName>\.nuget\packages\wixtoolset.sdk\4.0.0-preview.1\tools\net472\wix.exe" is invalid.

> It appears this is caused by the executable now being located in platform specific subdirectory. 

* Describe the behavior you expected and how it differed from the actual behavior.

> Successful wixlib project build.

**Comment by robmen** [comment]
Can you share a repository with a project that demonstrates this or share the .binlog from using `msbuild -bl` to get this error?

**Comment by dkalbertson** [comment]
[ThirdPartyComponents_Debug_x64_Build_2022-11-17T18_57_32.3025523-05_00.binlog.zip](https://github.com/wixtoolset/issues/files/10036601/ThirdPartyComponents_Debug_x64_Build_2022-11-17T18_57_32.3025523-05_00.binlog.zip)
@robmen Here's a bin log. The project is relatively small; I believe 3 wxs files. If this log doesn't help, let me know and I'll try to generate a sample project to replicate. Thanks!

**Comment by dkalbertson** [comment]
@robmen I've created a sample reproducible sample wix lib project on [Github: SampleWixLibV4Upgrade](https://github.com/dkalbertson/SampleWixLibV4Upgrade). The initial commit in the main repo is able to be built as a v3 WixLib. The next commit contains the changes after conversion using Heatwave that produces the error mentioned in the issue description.

**Comment by robmen** [comment]
Thank you very much. I pulled down your repo, and the .wixlib built on the first try for me. There is something unexpected about your machine. I wasn't able to discern anything on a quick look through the .binlog, so I'm going to have to add additional diagnostics to catch whatever case is unique on your machine (probably logging I should have added originally) and get a new .binlog.

This issue is next on my list to tackle.

**Comment by dkalbertson** [comment]
Thanks for the update. One potential issue (maybe) could be that on this machine, my username has a space in it. The machine wide cache of nuget packages resides in %UserProfile% (which for me resolves to C:\Users\daniel albertson\.nuget\packages). I'm not sure if that could cause this or not. However, that is one potential difference in our setups. Prior to the V4 upgrade, nuget packages were consumed via packages.config, not Packages Reference and, therefore, restored relative to the solution path that did not contain spaces (C:\Dev\temp\SampleWixLib\packages).

**Comment by dkalbertson** [comment]
@robmen, I did some additional testing this morning. It does appear to be related to the path of the nuget cache, having a space in it. I tested by adding an environment variable, NUGET_PACKAGES, that explicitly sets the nuget packages location to C:\Dev\Packages. I reopened the sample project/solution. I verified that the wix sdk and util extension were downloaded to the newly defined location and rebuilt the project/solution. It built successfully. To double check, I removed the environment variable, deleted the newly download packages and re-opened the project/solution now again using the default %userprofile% nuget packages location. After rebuilding, I got the build error again.
